src/Compilers/CSharp/Portable/Symbols/Metadata/PE/PEPropertySymbol.cs

```diff
@@ -867,8 +938,31 @@ internal override ObsoleteAttributeData ObsoleteAttributeData
         {
             get
             {
-                ObsoleteAttributeHelpers.InitializeObsoleteDataFromMetadata(ref _lazyObsoleteAttributeData, _handle, (PEModuleSymbol)(this.ContainingModule), ignoreByRefLikeMarker: false, ignoreRequiredMemberMarker: false);
-                return _lazyObsoleteAttributeData;
+                if (!_flags.IsObsoleteAttributePopulated)
```

Comment:
Implementation basically copied wholesale from `PEMethodSymbol`, with adjustments to `GetObsoleteDataFromMetadata` as appropriate.